pytorch_lightning/utilities/enums.py

```diff
@@ -111,3 +111,11 @@ class GradClipAlgorithmType(LightningEnum):
     """
     VALUE = 'value'
     NORM = 'norm'
+
+
+class AutoRestartBatchKeys(LightningEnum):
+    """
+    Defines special dictionary keys used to track sampler progress with multiple workers.
+    """
+
+    PL_SAMPLERS = "__pl_samplers__"
```

Comment:
I don't think there's value in creating an Enum just for this.

And, we don't have enums for all the keys saved on the checkpoints

Reply:
Why the double underscore before and after?

Reply:
This is required. 

This is used to make sure that when the CaptureIterableDataset return a batch, we can easily detect t it as it contains this key.

c.f: https://github.com/PyTorchLightning/pytorch-lightning/pull/8131/files#diff-ab48e8df50bc4933f169ead5982b38d36f21250a013c9aae7c4974968de43b35R473

Reply:
My point is, just use the `"pl_samplers"` key in the checkpoint